app/models/spree/order_decorator.rb

```diff
@@ -82,6 +83,18 @@ def products_available_from_new_distribution
     errors.add(:base, I18n.t(:spree_order_availability_error)) unless DistributionChangeValidator.new(self).can_change_to_distribution?(distributor, order_cycle)
   end
 
+  def using_guest_checkout?
+    require_email && !user.andand.id
```

Comment:
I had to look into the Spree code to find `require_email`. At first I thought it might be something Active Record provides. It's good to use that method here, but we need to keep that in mind during the Spree upgrade. It may change.

Reply:
Can you do a quick check to see if that's the case @mkllnk 🙏 ?

Reply:
If what is the case? So `require_email` is defined on Spree::Order to check if we are in a checkout step that requires an email. There are earlier steps that don't require an email, for example when you add stuff to your cart, an order is created and doesn't need an email.

Reply:
I meant if you could check the "it may change" part so that we know in advance, sorry

Reply:
Ah, the code changes slightly, but should be the same meaning. Compared to spree master:
```diff
-  return true unless new_record? or state == 'cart'
+  true unless new_record? || ['cart', 'address'].include?(state)
```

Same in Solidus master. All good. :-)

Reply:
@mkllnk is the plan to move to Solidus?

Reply:
There's a loooong way before we can even consider that decision :trollface:

Reply:
We don't have a plan for that stage yet. I just thought it would be interesting to know if a code change like this could affect a future decision like that. Just something to keep in mind. I think we will do the decision which fork to use when we updated to the latest spree version before the Solidus fork.